Source/OxyPlot/Foundation/DataVector.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            get
+            {
+                return Math.Sqrt((this.x * this.x) + (this.y * this.y));
+            }
+        }
+
+        /// <summary>
+        /// Gets the length squared.
+        /// </summary>
+        public double LengthSquared
+        {
+            get
+            {
+                return (this.x * this.x) + (this.y * this.y);
+            }
+        }
+
+        /// <summary>
+        /// Gets the x-coordinate.
+        /// </summary>
+        /// <value>The x-coordinate.</value>
+        public double X
+        {
+            get
+            {
+                return this.x;
+            }
+        }
+
+        /// <summary>
+        /// Gets the y-coordinate.
+        /// </summary>
+        /// <value>The y-coordinate.</value>
+        public double Y
+        {
+            get
+            {
+                return this.y;
+            }
+        }
+
+        /// <summary>
+        /// Implements the operator *.
+        /// </summary>
+        /// <param name="v">The vector.</param>
+        /// <param name="d">The multiplication factor.</param>
+        /// <returns>The result of the operator.</returns>
+        public static DataVector operator *(DataVector v, double d)
+        {
+            return new DataVector(v.x * d, v.y * d);
+        }
+
+        /// <summary>
+        /// Adds a vector to another vector.
+        /// </summary>
+        /// <param name="v">The vector to add to.</param>
+        /// <param name="d">The vector to be added.</param>
+        /// <returns>The result of the operation.</returns>
+        public static DataVector operator +(DataVector v, DataVector d)
+        {
+            return new DataVector(v.x + d.x, v.y + d.y);
+        }
+
+        /// <summary>
+        /// Subtracts one specified vector from another.
+        /// </summary>
+        /// <param name="v">The vector to subtract from.</param>
+        /// <param name="d">The vector to be subtracted.</param>
+        /// <returns>The result of operation.</returns>
+        public static DataVector operator -(DataVector v, DataVector d)
+        {
+            return new DataVector(v.x - d.x, v.y - d.y);
+        }
+
+        /// <summary>
+        /// Negates the specified vector.
+        /// </summary>
+        /// <param name="v">The vector to negate.</param>
+        /// <returns>The result of operation.</returns>
+        public static DataVector operator -(DataVector v)
+        {
+            return new DataVector(-v.x, -v.y);
+        }
+
+        /// <summary>
+        /// Normalizes this vector inplace.
+        /// </summary>
+        /// <remarks>
+        /// If the vector has zero length, then it is left unchanged.
+        /// </remarks>
+        public void Normalize()
+        {
+            double l = Math.Sqrt((this.x * this.x) + (this.y * this.y));
+            if (l > 0)
+            {
+                this.x /= l;
+                this.y /= l;
+            }
+        }
```

Comment:
I always have a bad feeling about mutable `struct`s. I think I would like it more if this returned a new struct.
Also this should use the `Length` property I reckon.